internal/envconfig/xds.go

```diff
@@ -99,3 +99,9 @@ var (
 	// C2PResolverTestOnlyTrafficDirectorURI is the TD URI for testing.
 	C2PResolverTestOnlyTrafficDirectorURI = os.Getenv(c2pResolverTestOnlyTrafficDirectorURIEnv)
 )
+
+// XDSRingHashLimit indicates the maximum ring size which defaults to 4096 but
+// may be overridden by setting the environment variable
+// "GRPC_XDS_RING_HASH_LIMIT".  A hard maximum limit of 8MB and minimum limit
```

Comment:
RING_HASH_LIMIT is somewhat ambiguous, since this limit really controls the max number of ring hash entries, and only indirectly controls memory (e.g. per-entry memory consumption could be reduced with a revised layout).

Note too the `A hard maximum limit of 8MB` should say something like `A hard maximum limit of 8 million entries, which can consume ~256MB`.

Reply:
> RING_HASH_LIMIT

Can you suggest something better?  I wanted to avoid something too verbose.  `XDS_RING_SIZE_LIMIT` is probably too vague, and `XDS_RING_HASH_RING_SIZE_LIMIT` is getting pretty long.  Is `XDS_RING_HASH_SIZE_LIMIT` still too ambiguous?

> Note too the A hard maximum limit of 8MB should say something like A hard maximum limit of 8 million entries, which can consume ~256MB

Well, it's apparently not 8 million either, it's 8,388,608.  Why did we choose that number (for Java) btw?

Reply:
Oh wait.. we actually NACK for configs > 8M.  So probably this is irrelevant to even mention here?

Reply:
In C++, their's a channel option called `GRPC_ARG_RING_HASH_LB_RING_SIZE_CAP`.

In Java, the global is `setRingSizeCap`.

So maybe use `GRPC_XDS_RING_HASH_CAP` ?

> Oh wait.. we actually NACK for configs > 8M. So probably this is irrelevant to even mention here?

Yeah I think it's not really relevant. Though I suppose we could mention that `GRPC_XDS_RING_HASH_CAP` does not influence the validation of max ring size configuration.

Reply:
> So maybe use `GRPC_XDS_RING_HASH_CAP`?

This is fine since it matches the other languages re: cap/limit, though I find "cap" to be less formal/precise than "limit".

Changed.

Reply:
Actually, should we remove "xds" from the name, too, since the LB policy name doesn't have "xds" in it?

Reply:
(I went ahead and did that in another commit; LMK.)